cogs/moderation.py

```diff
@@ -492,6 +572,184 @@ async def on_member_kick(self, target, entry):
         )
         await self.save_action(action)
 
+    @commands.hybrid_group(
+        aliases=("emergency-staff", "alert", "alert-staff"),
+        cooldown_after_parsing=True,
+        fallback="send",
+        invoke_without_subcommand=True,
+    )
+    @commands.cooldown(1, EMERGENCY_COOLDOWN_HOURS * 60 * 60, commands.BucketType.guild)  # Cooldown per guild
+    @commands.guild_only()
+    async def emergency(self, ctx: GuiduckContext, *, reason: str):
+        """Emergency command to alert staff members.
+
+        Do no abuse. Meant for use during emergencies that need immediate staff attention.
+        """
+
+        member = await self.bot.mongo.db.member.find_one({"_id": {"id": ctx.author.id, "guild_id": ctx.guild.id}})
+        if until := member.get("emergency_alert_banned_until"):
+            if until is True:
+                ctx.command.reset_cooldown(ctx)
+                return await ctx.send(
+                    "You've been permanently banned from issuing emergency staff alerts due to violation(s) of its rules. If you think that this was a mistake, please contact a staff member."
+                )
+            elif until is not None and until > (now := datetime.now(timezone.utc)):
+                ctx.command.reset_cooldown(ctx)
+                seconds = (until - now).total_seconds()
+                return await ctx.send(
+                    f"You've been banned from issuing emergency staff alerts for **{time.human_timedelta(timedelta(seconds=seconds))}** due to violation(s) of its rules."
+                )
+
+        guild = await self.bot.mongo.db.guild.find_one({"_id": ctx.guild.id})
+        role = ctx.guild.get_role(guild.get("emergency_alert_role")) if guild else None
+        if role is None:
+            ctx.command.reset_cooldown(ctx)
+            return await ctx.send(
+                "Emergency Staff Alert role not found for this guild. Please ask an Administrator to set one up.",
+                ephemeral=True,
+            )
+
+        number_staff = len(role.members)
+        confirm_embed = discord.Embed(
+            color=discord.Color.red(),
+            title="🚨 Emergency Staff Alert",
+            description=textwrap.dedent(
+                f"""
+                This command is designed for use in case of emergencies actively happening in our server(s) that need immediate staff attention. This will ping **{number_staff}** staff member{'' if number_staff == 1 else 's'} currently assigned to the {role.mention} role, and you will be assisted shortly.
+                """
+            ),
+        )
+        confirm_embed.add_field(
+            name="Are you sure that you want to send an Emergency Staff Alert for the following reason?",
+            value=reason,
+            inline=False,
+        )
+
+        rules_embed = discord.Embed(
```

Comment:
this embed can just be a constant i think

Reply:
Done!